junit-console/src/main/java/org/junit/gen5/console/tasks/XmlReportWriter.java

```diff
@@ -125,6 +126,17 @@ private void writeTestcase(TestIdentifier test, NumberFormat numberFormat, XMLSt
 		writer.writeEndElement();
 	}
 
+	private String getParentClassName(TestIdentifier test) {
+		// @formatter:off
+		return reportData.getTestPlan().getParent(test)
+				.flatMap(TestIdentifier::getSource)
+				.filter(JavaClassSource.class::isInstance)
+				.map(JavaClassSource.class::cast)
+				.map(source -> source.getJavaClass().getName())
+				.orElse("<unrooted>");
```

Comment:
Can we use the display name of the parent as a fallback? I think `<unrooted>` is really misleading. We can do it in a separate issue if you'd prefer.